src/main/java/ch/fhnw/algd1/converters/binary/BinConverter.java

```diff
@@ -2,14 +2,31 @@
 
 public class BinConverter {
 	public static String toString(int x) {
-		// TODO expect x to be in range [-128, 127], return string with 8 binary
-		// digits representing x in 2-complement
-		return "00000000";
+		String s = "";
+		if (x < 0) {
+			x += 128;
+			s += 1;
+		} else {
+			s += 0;
+		}
+
+		for (int i = 6; i >= 0; i--) {
+			s += (x >> i) & 1; // add the (i)th from right bit
```

Comment:
Hier gibt es jetzt einen kleinen "Betrug". Die Expression evaluiert zu einer Zahl (Wert 0 oder 1) und die wird dann implizit zu einem String umgewandelt. Da das allerdings für eine beliebige Zahl programmiert ist, läuft da so einiges im Hintergrund ab. Für den Fall hier könnten Sie das einfach durch eine explizite Fallunterscheidung abkürzen:
```suggestion
			s += ((x >> i) & 1) == 0 ? '0' : '1'; // add the (i)th from right bit
```